bodhi/messages/schemas/updates.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (C) 2018-2019 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+"""
+Message schema for Bodhi.
+
+Each message is defined as a Python class. For details, see `fedora-messaging
+<https://fedora-messaging.readthedocs.io/en/stable/>`_ documentation on
+messages.
+"""
+
+from .base import BodhiMessage
+
+
+class UpdatesSyncMessage(BodhiMessage):
+    """Base class for updates sync messages."""
+
+    @property
+    def summary(self) -> str:
+        """
+        Return a short, human-readable representation of this message.
+
+        This should provide a short summary of the message, much like the subject line
+        of an email.
+
+        Returns:
+            A summary for this message.
+        """
+        tmpl = (
+            'New {product} {release} {repo} content synced out '
+            '({bytesize} changed with {deleted} files deleted)'
+        )
+        return tmpl.format(
+            product=self._product,
+            release=self._body["release"],
+            repo=self._body["repo"],
+            bytesize=self._body["bytes"],
+            deleted=self._body["deleted"],
+        )
+
+    @property
+    def url(self) -> str:
+        """
+        Return a URL to the action that caused this message to be emitted.
+
+        Returns:
+            A relevant URL.
+        """
+        link = "https://download.fedoraproject.org/pub/"
+        repo = self._body['repo']
+        product = self._product.lower()
+        link += product + "/"
+
+        if product == 'fedora':
+            if repo == 'atomic':
+                link += "linux/atomic/"
+            else:
+                link += "linux/updates/"
+
+        if repo.endswith('testing'):
+            link += "testing/"
+
+        link += self._body['release'] + "/"
+        return link
+
+
+class UpdatesEpelSyncV1(UpdatesSyncMessage):
+    """Sent when EPEL updates are synced."""
+
+    topic = "bodhi.updates.epel.sync"
+    _product = "EPEL"
+
+
+class UpdatesFedoraSyncV1(UpdatesSyncMessage):
+    """Sent when Fedora updates are synced."""
+
+    topic = "bodhi.updates.fedora.sync"
```

Comment:
Despite the topic, I don't think the messages in this file are sent by Bodhi. They seem to be sent by the script that does the sync. Note the username in this one, for example:

https://apps.fedoraproject.org/datagrepper/id?id=2019-a5d230e5-e086-4588-ac16-48b14335c4b0&is_raw=true&size=extra-large

When Bodhi sends messages, the username is apache:

https://apps.fedoraproject.org/datagrepper/id?id=2019-cd7f6213-1c0e-465e-9793-5c4cec6a4607&is_raw=true&size=extra-large

I'm conflicted about whether we should have these schemas in Bodhi or not, since Bodhi doesn't actually send them and thus isn't actually in control of their format. What do you think?